File: core/src/main/java/com/cc/core/actions/accessibility/impl/BackAction.java
package com.cc.core.actions.accessibility.impl;

import com.cc.core.actions.ActionResult;
import com.cc.core.actions.accessibility.AccessibilityAction;

public class BackAction extends AccessibilityAction {
    @Override
    public ActionResult execute(Object... args) {
        boolean result = systemBack();
        return result ? ActionResult.successResult() : ActionResult.failedResult("can not execute back");
    }

    @Override
    public String key() {
        return "back";
    }
}

File: core/src/main/java/com/cc/core/actions/accessibility/impl/BackAction.kt
package com.cc.core.actions.accessibility.impl

import com.cc.core.actions.ActionResult
import com.cc.core.actions.accessibility.AccessibilityAction

class BackAction : AccessibilityAction() {
    override fun execute(actiongId : String, vararg args: Any?): ActionResult? {
        val result = systemBack()
        return if (result) ActionResult.successResult(actiongId) else ActionResult.failedResult(actiongId, "can not execute back")
    }

    override fun key(): String {
        return "back"
    }
}